42/README.md

```diff
@@ -70,14 +71,133 @@ The above process ensure the node keeps validating and accepting new blocks whil
+
+#### 3. Start the new chain
+
+Validators have to install and launch the new version of Tendermint. They need to prepare a new `genesis.json`. The new genesis file has to contain the validator set, the application hash (both at the migration height), and the identifier of the new Tendermint chain (`chain_id`).
+
+The application hash is calculated by computing the Merkle tree root of the current UTXO. Note that hash calculation is a subject to design in a separate BEP and can be changed in the future.
+
+`chain_id` is generated and stored by BigchainDB upon conclusion of the migration election. When Tendermint sends this ID as part of the `InitChain` ABCI request, BigchainDB understands that the user has switched to a new Tendermint version so BigchainDB switches to accept new transactions and blocks.
+
+To offer a convenient way to get the data, we propose a CLI command:
+
+```
+$ bigchaindb status
+```
+
+Its output can be extended over time to solve different purposes. For a start, it has to contain the latest known list of validators, the latest known app hash, and the identifier of the Tendermint chain.
+```
+{
+    "chain_id": "...",
+    "validators": [{
+        "pub_key": {
+            "type": "...",
+            "value": "..."
+        },
+        "power": "...",
+        "name": "..."
+    },
+    "app_hash": "..."
+}
+```
+
+Validators upgrading Tendermint are supposed to execute the command above and copy the validators, the app hash, and the chain ID into their `genesis.json`.
+
+Validators joining the network after the migration take the genesis file from existing validators, as usual.
+
+#### 4. Join the network after a migration
+
+When a validator joins the network after a migration, it does not suffice for him to know `genesis.json` - he also needs the archive of the old chain.
+There might be more than one migration, so the initial chain might go from height 0 to 33277, the second chain might go to height 88234, the third chain can go from 88234 up to the recent height. In this case the validator has to get an archive containing archived blocks from height 0 to height 88234.
+
+Such a validator needs to take a snapshot of the old chain from an existing member of the network. The snapshot can be created via a CLI command:
+
+```
+$ bigchaindb migration create-snapshot
+```
+
+The command creates the `bigchaindb.snapshot` file that needs to be transferred to the new member. The snapshot contains all BigchainDB collections to restore the old chain from.
+
+The new validator needs to start BigchainDB without Tendermint and apply the snapshot:
+
+```
+$ bigchaindb apply-snapshot <snapshot>
+```
+
+After the snapshot is applied, Tendermint can be started.
+
+Note that although the new node can join the network and work to some extent without applying the snapshot, it is not able to properly validate transactions so applying the snapshot is a must.
```

Comment:
What effect will this have on the network? I suppose this node be treated as byzantine node?

Reply:
Yes, it will. Without sufficient data, it may claim invalid transactions to be valid.